mattermost/mattermost-plugin-jira
# [MM-891]: Added reporter and assignee fields in subscription filters

#### Summary
- Added `reporter` and `assignee` fields to subscriptions modal to filter subscription notifications.
- Added the `reporter` field to the `issue create dialog modal` if the project supports it.

This PR is a recreation of https://github.com/mattermost/mattermost-plugin-jira/pull/890 since the creator of https://github.com/mattermost/mattermost-plugin-jira/pull/890 has been inactive for a long time.

#### Screenshots
![image](https://user-images.githubusercontent.com/107976929/202347087-2d4d483d-bba9-4c1a-b791-c0ae3683e4c6.png)
![image](https://user-images.githubusercontent.com/107976929/202347116-8ccb1453-69b5-4fc5-a57f-ed4a6f9d0848.png)

#### What to test
- Set up Jira Oauth & Webhook and connect to the Jira instance
- Create a subscription and add a filter for reporter/assignee
- Create an issue on Jira similar to the subscription created but with a different reporter/assignee, you should not receive the subscription update on the channel if you had chosen `includes` in the filter(similarly for other filter types).
-  Create an issue on Jira similar to the subscription created but with the same reporter/assignee, you should receive the subscription update on the channel if you had chosen `includes` in the filter.

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-plugin-jira/issues/891
